Graylog2/graylog2-server
# Protect used content pack entities from uninstall

## Description
This PR will protect used content pack entities from removal through uninstall if they are either
still in use bye a other content pack installation, or were on the system before installing a content
pack.

## Motivation and Context
If a content pack would delete all its entities, other installations would stop working, since we 
can't install the same grok pattern twice.

## How Has This Been Tested?
1. Create a grok pattern (as it exists in the default grok pattern)
2. installed default pattern
3. install a content pack using a pattern from the default content pack
4. uninstall the last content pack
5. check that grok pattern is still there
6. Uninstall default grok patterns
7 check that grok pattern is still there 

Fixes: #5399 

## Types of changes

## Checklist:
